refactoring/initial_two_classes.py

```diff
@@ -32,17 +32,27 @@ def add_person(self, name, age, job):
 
     def number_of_connections(self, name):
         """Find the number of connections that a person in the group has"""
-        pass
+        return len(self.connections[name])
 
     def connect(self, name1, name2, relation, reciprocal=True):
         """Connect two given people in a particular way.
         Optional reciprocal: If true, will add the relationship from name2 to name 1 as well
         """
-        pass
+        try:
+            self.connections[name1] += [(name2, relation)]
+        except:
+            self.connections[name1] = [(name2, relation)]
+        if reciprocal:
+            self.connect(name2, name1, relation, False)
```

Comment:
Ah interesting, I thinik our example we the pairs of connections but this works too. Again could have some guard clauses to make sure that the names only existing people are being used and they they already know 
```suggestion
        if not self.contains(name1):
            raise ValueError(f"I don't know who {name1} is.")
        if not self.contains(name2):
            raise ValueError(f"I don't know who {name2} is.")
        try:
            self.connections[name1] += [(name2, relation)]
        except:
            self.connections[name1] = [(name2, relation)]
        if reciprocal:
            self.connect(name2, name1, relation, False)
```